src/System.Private.CoreLib/shared/System/BitOps.cs

```diff
@@ -34,5 +41,459 @@ public static int TrailingZeroCount(int matches)
             0, 1, 28, 2, 29, 14, 24, 3, 30, 22, 20, 15, 25, 17, 4, 8,
             31, 27, 13, 23, 21, 19, 16, 7, 26, 12, 18, 6, 11, 5, 10, 9
         };
+
+        #region LZCNT
+
+        // This method moved & renamed from the following location
+        // https://raw.githubusercontent.com/dotnet/corefx/62c70143cfbb08bbf03b5b8aad60c2add84a0d9e/src/Common/src/CoreLib/System/Number.BigInteger.cs
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static uint LeadingZeroCount_Number(uint value)
+        {
+            return 32 - CountSignificantBits(value);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static uint LeadingZeroCount_Number(ulong value)
+        {
+            return 64 - CountSignificantBits(value);
+        }
+        
+        private static readonly uint[] s_MultiplyDeBruijnBitPosition = new uint[]
+        {
+            0, 9, 1, 10, 13, 21, 2, 29, 11, 14, 16, 18, 22, 25, 3, 30,
+            8, 12, 20, 28, 15, 17, 24, 7, 19, 27, 23, 6, 26, 5, 4, 31
+        };
+
+        // This is a similar method from a different location.
+        // It is shown here in its entirety, to show ubiquity of such code.
+        // It has been renamed so that the class compiles.
+        // Duplicates will be consolidated and optimized later.
+        // https://raw.githubusercontent.com/dotnet/corefx/151a6065fa8184feb1ac4a55c89752342ab7c3bb/src/Common/src/CoreLib/System/Decimal.DecCalc.cs
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int LeadingZeroCount_DecCalc(uint value)
```

Comment:
I have not done any analysis yet so see if duplicates have the exact same behavior as their brethren. That will come later with units.